app/models/proposal.rb

```diff
@@ -90,7 +98,7 @@ def closed_state
     def open_state
       return nil if pr_closed?
       return "dead" if too_old?
-      return "blocked" if blocked?
+      return "blocked" if blocked? || (cla_required? && !cla_accepted?)
```

Comment:
This logic happens quite a lot. I feel like there's a combined method like `User#needs_to_sign_cla?` that would reduce the places where both have to be checked.